.github/workflows/deployment.yml

```diff
@@ -23,104 +43,144 @@ jobs:
-        get-childitem -Path build/distribution/* | rename-item -NewName {$_.name -replace "portable","${{ steps.gitversion.outputs.Major }}.${{ steps.gitversion.outputs.Minor }}-portable"}
-      shell: pwsh
-    - name: Upload to GitHub workflow artifacts store
-      uses: actions/upload-artifact@master
-      with:
-        name: JabRef-${{ matrix.displayName }}
-        path: build/distribution
-    - name: Upload to builds.jabref.org
-      uses: garygrossgarten/github-action-scp@release
-      with:
-        local: build/distribution
-        remote: www/${{ steps.gitversion.outputs.branchName }}
-        host: builds.jabref.org
-        username: builds_jabref_org
-        privateKey: ${{ secrets.buildJabRefPrivateKey }}
-        port: 9922
+          url = "${{ matrix.jpackageDownload }}"
+          tmpfile, headers = urlretrieve(url)
+          if (url.endswith("tar.gz")):
+            tar = tarfile.open(tmpfile)
+            tar.extractall()
+            tar.close()
+          elif (url.endswith("zip")):
+            zip = zipfile.ZipFile(tmpfile)
+            zip.extractall()
+            zip.close()
+        shell: python
+      - name: Build runtime image
+        run: ./gradlew -PprojVersion="${{ steps.gitversion.outputs.AssemblySemVer }}" -PprojVersionInfo="${{ steps.gitversion.outputs.InformationalVersion }}" jlinkZip
+      - name: Build installer
+        run: |
+          export BADASS_JLINK_JPACKAGE_HOME="${GITHUB_WORKSPACE}${{ matrix.jdk14Path }}"
+          ./gradlew -PprojVersion="${{ steps.gitversion.outputs.AssemblySemVer }}" -PprojVersionInfo="${{ steps.gitversion.outputs.InformationalVersion }}" jpackage
+        shell: bash
+      - name: Package application image
+        run: ${{ matrix.archivePortable }}
+        shell: bash
+      - name: Build and publish snap
+        if: matrix.os == 'ubuntu-latest' && steps.gitversion.outputs.branchName == 'master'
+        env:
+          SNAPCRAFT_LOGIN_FILE: ${{ secrets.SNAPCRAFT_LOGIN_FILE }}
+        run: |
+            mkdir .snapcraft && echo ${SNAPCRAFT_LOGIN_FILE} | base64 --decode --ignore-garbage > .snapcraft/snapcraft.cfg
+            docker run -v $(pwd):$(pwd) -t lyzardking/snapcraft-bionic sh -c "apt update -qq && cd $(pwd) && snapcraft && mv jabref*.snap build/distribution/ && snapcraft push build/distribution/jabref*.snap --release edge || true"
+        shell: bash
+      - name: Rename files
+        run: |
+          get-childitem -Path build/distribution/* | rename-item -NewName {$_.name -replace "${{ steps.gitversion.outputs.AssemblySemVer }}","${{ steps.gitversion.outputs.Major }}.${{ steps.gitversion.outputs.Minor }}"}
+          get-childitem -Path build/distribution/* | rename-item -NewName {$_.name -replace "portable","${{ steps.gitversion.outputs.Major }}.${{ steps.gitversion.outputs.Minor }}-portable"}
+        shell: pwsh
+      - uses: actions/upload-artifact@master
+        with:
+          name: ${{ matrix.os }}-distribution
```

Comment:
`matrix.displayName` has the advantage that it's stable even if we should decide to choose a different ubunto version